workflows/msmc.py

```diff
@@ -28,40 +28,34 @@ def getIntervalOverlap(intA, intB):
     return max(0, min(intA.right, intB.right) - max(intA.left, intB.left))
 
 
-def write_msmc_file(path, output, num_sampled_genomes_msmc, mask_file=None):
+def write_msmc_file(path, output, num_sampled_genomes_msmc, mask_intervals=None):
     """
     take one .trees file and write out
     path.multihep.txt which acts as a single input to msmc
 
     This seems hacky atm, but let's getting working then hash out
     the details
     """
-    # make the mask intervalindex?
-    if mask_file:
-        mask_table = pd.read_csv(mask_file, sep="\t", header=None)
-        mask_dict = {}
-        for key in mask_table[0].unique():
-            sub = mask_table[mask_table[0] == key]
-            mask_dict[key] = pd.IntervalIndex.from_arrays(sub[1], sub[2])
-        print(mask_dict)
 
     for sample_size in num_sampled_genomes_msmc:
         ts = prune_tree_sequence(path, sample_size)
         dirr = os.path.dirname(path)
         filen = os.path.basename(path)
         sep = filen.split(".")
-        chrom = sep[0].split("_")[0]
+        chrom = sep[0].split("_")[1] # fixing sim_ name
         sep.insert(0, str(sample_size))
         fi = open(output, "w")
         prev = 0
-        if mask_file:
+        if type(mask_intervals) != None:
+            print(mask_intervals)
+            mask_intervals = pd.IntervalIndex.from_arrays(mask_intervals[:,0], mask_intervals[:,1])
             for var in ts.variants():
```

Comment:
variants are in 1-to-1 correspondence with sites, and so always have unique positions: the `cur` and `prev` stuff here is unnecessary.

Reply:
sorry, I mean the `if cur > prev` condition

Reply:
ah except for the first position, if it's at site 0, so never mind - it's harmless